appium/webdriver/extensions/action_helpers.py

```diff
@@ -42,32 +42,23 @@ def scroll(self: T, origin_el: WebElement, destination_el: WebElement, duration:
         Returns:
             Union['WebDriver', 'ActionHelpers']: Self instance
         """
-
         # XCUITest x W3C spec has no duration by default in server side
         if duration is None:
             duration = 600
 
-        # w3c action requires seconds
-        duration_sec = duration / 1000
+        touch_input = PointerInput(interaction.POINTER_TOUCH, "touch")
 
         actions = ActionChains(self)
-        actions.w3c_actions = ActionBuilder(self, mouse=PointerInput(interaction.POINTER_TOUCH, "touch"))
-        dest_el_rect = destination_el.rect
+        actions.w3c_actions = ActionBuilder(self, mouse=touch_input)
 
         # https://github.com/SeleniumHQ/selenium/blob/3c82c868d4f2a7600223a1b3817301d0b04d28e4/py/selenium/webdriver/common/actions/pointer_actions.py#L83
-        if duration_sec is None:
-            actions.w3c_actions.pointer_action.move_to(origin_el)
-            actions.w3c_actions.pointer_action.pointer_down()
-            actions.w3c_actions.pointer_action.move_to_location(dest_el_rect['x'], dest_el_rect['y'])
-            actions.w3c_actions.pointer_action.release()
-            actions.perform()
-        else:
-            actions.w3c_actions.pointer_action.move_to(origin_el)
-            actions.w3c_actions.pointer_action.pointer_down()
-            actions.w3c_actions.pointer_action.pause(duration_sec)
-            actions.w3c_actions.pointer_action.move_to_location(dest_el_rect['x'], dest_el_rect['y'])
-            actions.w3c_actions.pointer_action.release()
-            actions.perform()
+        actions.w3c_actions.pointer_action.move_to(origin_el)
+        actions.w3c_actions.pointer_action.pointer_down()
+        # setup duration for second move only, assuming duration always has atleast default value
+        actions.w3c_actions = ActionBuilder(self, mouse=touch_input, duration=duration)
```

Comment:
I don't understand the logic here - do we just reassign `actions.w3c_actions` property? what is then the point of having previous calls?

Reply:
Previous actions are in `touch_input`, so this one just updates `duration`.
`actions.w3c_actions.pointer_action._duration = duration` can be the alternative of the `ActionBuilder(self, mouse=touch_input, duration=duration)` to append actions in `touch_input` with the duration.

```
>>> touch_input
<selenium.webdriver.common.actions.pointer_input.PointerInput object at 0x108c750f0>
>>> touch_input.actions
[{'type': 'pointerDown', 'duration': 0, 'button': 0}, {'type': 'pointerDown', 'duration': 0, 'button': 0}, {'type': 'pointerUp', 'duration': 0, 'button': 0}]
```

Reply:
i think you can also call the `create_xxx` methods on touch_input itself, like `touch_input.create_pointer_move()`, right?